doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,364 @@
+argument of ``train_func`` and set the corresponding environment variable in the ``train_func``.
+
+For accessing config file, one needs to ensure that the config file is accessible to all nodes.
+One way to do this is by setting up a shared storage. Another way is to save a copy in each node.
+
+Setting up shared file system
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This involves setting up a network filesystem accessible to all nodes in the cluster, 
+e.g. AWS EFS or Google Cloud Filestore.
+
+Step 2: Configure and start the run 
+-----------------------------------
+
+This usually concerns picking an identifier for the run and associating it with a project.
+Please refer to tracking libraries' own documentation for semantics. 
+
+.. tip::
+    
+    When performing **fault-tolerant training** with auto-restoration, make sure that you use a 
+    consistent id to configure all tracking runs that logically belong to the same training run.
+    One way to acquire an unique id is through 
+    :meth:`ray.train.get_context().get_trial_id() <ray.train.TrainContext.get_trial_id>`.
+
+    .. code-block:: python
+
+        import ray
+        from ray.train import ScalingConfig, RunConfig, FailureConfig
+        from ray.train.torch import TorchTrainer
+
+        def train_func(config):
+            if ray.train.get_context().get_world_rank() == 0:
+                wandb.init(id=ray.train.get_context().get_trial_id())
+            ...
+
+        trainer = TorchTrainer(
+            train_func, 
+            run_config=RunConfig(failure_config=FailureConfig(max_failures=3))
+        )
+
+        trainer.fit()
             
-            - `WandbLogger(id=UNIQUE_ID)`
-            - `CometLogger(experiment_key=UNIQUE_ID)`
-            - `MLFlowLogger(run_id=UNIQUE_ID)`
 
+Step 3: Log
+-----------
 
-        
-Automatic setup methods
-~~~~~~~~~~~~~~~~~~~~~~~
-Ray Train provides utilities for common experiment tracking libraries to automatically
-setup an experiment with the training run name and ID used by Ray Train. It also
-initializes the library in a way that only the rank 0 worker reports the results per
-default.
+You can customize within ``train_func`` how to log parameters, metrics, models, or media contents 
+just as you would with a single process training script. 
+You can also use native integrations that a particular tracking framework has with 
+specific training frameworks, for example ``mlflow.pytorch.autolog()``, 
+``lightning.pytorch.loggers.MLFlowLogger`` etc. 
+
+Step 4: Finish the run
+----------------------
+
+This step ensures that all logs are synced to the tracking service. Depending on the implementation of 
+various tracking libraries, sometimes logs are first stored locally before they are synced to the tracking 
+service. When training is finished, the node that the training worker resided on could be turned down by 
+ray autoscaler. Finishing the run makes sure that all logs are synced by the time training workers exit. 
```

Comment:
```suggestion
service. Finishing the run makes sure that all logs are synced by the time training workers exit. 
```

There's no need to mention the node/autoscaler. I think more technically this happens when the Ray Actor terminates, but the important part for the user to know is that this will sync all the data before the training finishes.